### ISSUE TITLE
Select onInput doesn't function in Microsoft Edge

### ISSUE THREAD

**Issue by kamilgrzegorczyk** [preactjs/preact#2331]
Hello,

Browser: Microsoft Edge 44
Preact 10.3.1

Way to reproduce:
- visit the url https://preactjs.com/guide/v10/forms#select-input
- run the code in REPL
- the value is not getting updated (when you select any of the options)

Expected:
The value is updated properl and onInput callback is getting fired.

**Comment by marvinhagemeister** [comment]
Can't reproduce the issue in the chromium based versions of edge. Can confirm that the issue occurs on older versions with the Chakra engine.

**Comment by kamilgrzegorczyk** [comment]
Yeah, this is also my observation - new Edge engine does not have that issue. Unfortunately Microsoft Edge 44 was released just few months ago so it's hard to treat it as an unsupported browser.

**Comment by marvinhagemeister** [comment]
We never said that we don't support edge. In fact the oldest version we officially support is Internet Explorer 11. This is definitely a bug in Preact.

**Comment by kamilgrzegorczyk** [comment]
I believe that this might be due to the fact that Microsoft Edge does not support the onInput event on the `<select>` tags. I have used onchange there.

Source:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event
> Not supported on select, checkbox, or radio inputs.

**Comment by developit** [comment]
I think we can find a way to patch this since we're already special-casing onChange.
